packages/medusa/src/services/discount.ts

```diff
@@ -301,14 +302,55 @@ class DiscountService extends TransactionBaseService {
       if (!discount) {
         throw new MedusaError(
           MedusaError.Types.NOT_FOUND,
-          `Discount with code ${discountCode} was not found`
+          `Discounts with code ${discountCode} was not found`
         )
       }
     }
 
     return discount
   }
 
+  /**
+   * List all the discounts corresponding to the given codes
+   * @param discountCodes - discount codes of discounts to retrieve
+   * @param config - the config object containing query settings
+   * @return the discounts
+   */
+  async listByCodes(
+    discountCodes: string[],
+    config: FindConfig<Discount> = {}
+  ): Promise<Discount[]> {
+    const manager = this.manager_
+    const discountRepo = manager.getCustomRepository(this.discountRepository_)
+
+    const normalizedCodes = discountCodes.map((code) =>
+      code.toUpperCase().trim()
+    )
+
+    let query = buildQuery(
+      { code: In(normalizedCodes), is_dynamic: false },
```

Comment:
isn't it possible to bring `is_dinamic` true and false within the same query?

Reply:
I didn't want to change that behavior but it seems that they must all be found in one and if they are not all in one then they are found in the other value. Ideally, we would just find them by code but then we can have a mix of dynamic of not and I don't know right away if there is any impact to consider while doing that. This is why I decided to proceed the same way as the retrieve

Reply:
yeah apparently there is no special treatment for `is_dynamic` here. but ok, we can update that in another PR

Reply:
this should be enought wdyt @carlos-r-l-rodrigues and same for the retrieve

```ts
async listByCodes(
    discountCodes: string[],
    config: FindConfig<Discount> = {}
  ): Promise<Discount[]> {
    const manager = this.manager_
    const discountRepo = manager.getCustomRepository(this.discountRepository_)

    const normalizedCodes = discountCodes.map((code) =>
      code.toUpperCase().trim()
    )

    const query = buildQuery({ code: In(normalizedCodes) }, config)
    const discounts = await discountRepo.find(query)

    if (discounts?.length !== discountCodes.length) {
      throw new MedusaError(
        MedusaError.Types.NOT_FOUND,
        `Discounts with code [${normalizedCodes.join(", ")}] was not found`
      )
    }

    return discounts
  }
  ```

Reply:
@olivermrbl was there any reason why we were querying is_dynamic: false and if there is not then is_dynamic: true? why don't just query by code and return it whenever it is dynamic or not since the code is unique it can only exists in one or the other side right?

Reply:
> yeah apparently there is no special treatment for `is_dynamic` here. but ok, we can update that in another PR

ahah already changed it on my side, just waiting oli confirmation to be sure there is no unforeseen things

Reply:
I think it's fine. The field is non-nullable on the DB, so the behavior is bring everything.

Reply:
Makes sense! Nice catch @carlos-r-l-rodrigues 👍